samcli/commands/local/lib/swagger/parser.py

```diff
@@ -123,11 +128,27 @@ def get_authorizers(self, event_type: str = Route.API) -> Dict[str, Authorizer]:
             )
 
             validation_expression = authorizer_object.get(SwaggerParser._AUTHORIZER_LAMBDA_VALIDATION)
-            if event_type != Route.HTTP:
+            if event_type == Route.HTTP and validation_expression:
                 validation_expression = None
 
                 LOG.warning(
-                    "Validation expressions is only available on Rest APIs, ignoring for Lambda authorizer '%s'",
+                    "Validation expressions is only available on REST APIs, ignoring for Lambda authorizer '%s'",
+                    auth_name,
+                )
+
+            enable_simple_response = authorizer_object.get(SwaggerParser._AUTHORIZER_SIMPLE_RESPONSES, False)
+            if (
+                event_type == Route.API
+                and enable_simple_response
+                or event_type == Route.HTTP
+                and payload_version != LambdaAuthorizer.PAYLOAD_V2
+                and enable_simple_response
+            ):
```

Comment:
Anyway to simplify this if statements group?

Reply:
Oh yup, looking back at this statement, its not to follow at first glance 😞 

Will make this change in this PR: `https://github.com/aws/aws-sam-cli/pull/4938`